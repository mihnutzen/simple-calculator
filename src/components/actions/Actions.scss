@import 'src/styles/variables';

.Actions {
  display: grid;
  grid-template-columns: repeat($numbers-grid-columns, 1fr);
  grid-template-rows: repeat(1, $height);

  border-bottom: 1px solid #666;
  grid-gap: 1px;
  background-color: #666;

  > button {
    background: #999;
    line-height: $height;
  }
}