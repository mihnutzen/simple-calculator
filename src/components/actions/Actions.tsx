import React from 'react';

import { ActionsList, ActionsSigns } from '../../types/calculator';

import './Actions.scss';

const CLEAR = {
  key: ActionsList.Clear,
  sign: ActionsSigns.Clear,
  helper: 'Clear (Esc)'
};

const NEGATE = {
  key: ActionsList.Negate,
  sign: ActionsSigns.Negate,
  helper: 'Negate the displayed value'
};

const PERCENT = {
  key: ActionsList.Percent,
  sign: ActionsSigns.Percent,
  helper: 'Percent (or press %)'
}

const ACTIONS = [CLEAR, NEGATE, PERCENT];

interface ActionsProps {
  onAction: (act: ActionsList) => void;
}

const Actions = ({ onAction }: ActionsProps) => (
  <div className="Actions">
      {ACTIONS.map(({ key, sign, helper }) => (
        <button onClick={() => onAction(key)} key={key} title={helper}>{sign}</button>
      ))}
  </div>
);

export default Actions;