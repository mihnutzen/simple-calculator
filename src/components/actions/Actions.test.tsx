import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Actions from './Actions';

describe('The Actions component', () => {
  it('should render correctly', () => {
    const actionMock = jest.fn();

    const { asFragment } = render(<Actions onAction={actionMock} />);
    const firstRender = asFragment()

    expect(firstRender).toMatchSnapshot();
  });

  it('should trigger onAction with payload', () => {
    const actionMock = jest.fn();

    const { getByText } = render(<Actions onAction={actionMock} />);

    fireEvent.click(getByText(/AC/))

    expect(actionMock).toHaveBeenCalledWith('clear');
  });
});
