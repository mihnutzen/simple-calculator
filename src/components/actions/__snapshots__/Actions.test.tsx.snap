// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Actions component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="Actions"
  >
    <button
      title="Clear (Esc)"
    >
      AC
    </button>
    <button
      title="Negate the displayed value"
    >
      Â±
    </button>
    <button
      title="Percent (or press %)"
    >
      %
    </button>
  </div>
</DocumentFragment>
`;
